solr/core/src/java/org/apache/solr/core/SolrConfig.java

```diff
@@ -393,14 +393,22 @@ private SolrConfig(
   }
 
   private IndexSchemaFactory.VersionedConfig readXml(SolrResourceLoader loader, String name) {
+    InputStream in = null;
     try {
-      ResourceProvider rp = new ResourceProvider(loader, name);
+      in = loader.openResource(name);
```

Comment:
Can we use try-with-resources pattern here?

Reply:
I think we cannot, unfortunately; because in most cases the InputStream should have been closed properly by the parser before reaching our `finally` block, so we're really relying on the "exception-swallowing" quality of `IOUtils.closeQuietly()`.

Reply:
I think IndexSchemaFactory#loadConfig has the same pattern here? at least using `loader.openResource(name)` without closing it?

https://github.com/apache/solr/pull/987/files#diff-5965edf45e1dec49fea047168b81ddffbb530f5db8875b41627950773d63b3a6R138

Reply:
I went ahead and used try-with-resources (b6487869eec05cad5478fea05fc8bf224ccb8129) where possible (it's not possible in contexts where the InputStream is opened in less straightforward ways).

@risdenk in cd4923ff78ae43d80b2b43029ff282e5322a8e13 I also addressed the case you identified in IndexSchemaFactory, and followed up the stack trace to address a similar issue in ManagedIndexSchemaFactory.

As a consequence of the ManagedIndexSchemaFactory change, this PR's a bit easier to review ignoring whitespace: https://github.com/apache/solr/pull/1302/files?w=1